index.js

```diff
@@ -265,15 +272,18 @@ exports.extract = function (cwd, opts) {
 
     mkdirp(path.dirname(name), {fs: xfs}, function (err) {
       if (err) return next(err)
-      if (header.type === 'symlink') return onlink()
 
-      if (header.type !== 'file') {
-        if (strict) return next(new Error('unsupported type for ' + name + ' (' + header.type + ')'))
-        stream.resume()
-        return next()
+      switch (header.type) {
+        case 'file'   : return onfile()
+        case 'link'   : return onlink()
+        case 'symlink': return onsymlink()
```

Comment:
Previously for the `symlink` it was not being resumed, nor here or at the `onlink` function. Was it a bug? Should I add a resume on `onlink` and `onsymlink` functions?